Heat_method_3/include/CGAL/Heat_method_3/Surface_mesh_geodesic_distances_3.h

```diff
@@ -480,7 +480,8 @@ class Surface_mesh_geodesic_distances_3
   template<class VertexDistanceMap>
   void estimate_geodesic_distances(VertexDistanceMap vdm)
   {
-    CGAL_precondition(!internal::has_degenerate_faces(triangle_mesh(), vertex_point_map()));
+    CGAL_precondition(
+      !CGAL::Heat_method_3::internal::has_degenerate_faces(triangle_mesh(), vertex_point_map()));
```

Comment:
The error reported by Travis was:
```
 /home/travis/build/CGAL/cgal/Installation/../Heat_method_3/include/CGAL/Heat_method_3/Surface_mesh_geodesic_distances_3.h:483:24: error: no member named 'has_degenerate_faces' in namespace 'CGAL::Heat_method_3::internal'; did you mean 'Polygon_mesh_processing::degenerate_faces'?
CGAL_precondition(!internal::has_degenerate_faces(triangle_mesh(), vertex_point_map()));
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

That patch will not change anything: the compiler does not see the declaration of `CGAL::Heat_method_3::internal::has_degenerate_faces` because it is declared *after* in the same file.

You can either move the declaration/definition, or forward-declare it.